src/test/java/seedu/address/logic/commands/UnmatchCommandTest.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    @Test
+    public void constructor_nullIndex_throwsNullPointerException() {
+        Index validIndex = Index.fromOneBased(1);
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, null));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, validIndex));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(validIndex, null));
+    }
+
+    @Test
+    public void execute_validContactAndJob_unmatchSuccessful() throws Exception {
+        Index contactIndex = INDEX_FIRST_PERSON;
+        Index jobIndex = INDEX_FIRST_JOB;
+
+        // First, set up the model with an existing match
+        setModelAfterMatching(model, contactIndex, jobIndex);
+
+        UnmatchCommand unmatchCommand = new UnmatchCommand(contactIndex, jobIndex);
+
+        // Set up the expected model state after unmatching
+        Person contactInExpectedModel = expectedModel.getFilteredPersonList().get(contactIndex.getZeroBased());
+        Person unmatchedContact = new PersonBuilder(contactInExpectedModel).withoutMatch().build();
+        expectedModel.setPerson(contactInExpectedModel, unmatchedContact);
+
+        // Format the expected success message
+        String expectedMessage = String.format(UnmatchCommand.MESSAGE_UNMATCH_SUCCESS,
+                Messages.format(unmatchedContact), Messages.format(expectedModel
+                        .getFilteredJobList().get(jobIndex.getZeroBased())));
+
+        // Verify the command execution and resulting state
+        assertCommandSuccess(unmatchCommand, model, expectedMessage, expectedModel);
+    }
+
+
+    @Test
+    public void execute_contactIndexOutOfRange_failure() {
+        UnmatchCommand unmatchCommand = new UnmatchCommand(INDEX_OUT_OF_BOUNDS, INDEX_FIRST_JOB);
+        assertCommandFailure(unmatchCommand, model, Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_jobIndexOutOfRange_failure() {
+        UnmatchCommand unmatchCommand = new UnmatchCommand(INDEX_FIRST_PERSON, INDEX_OUT_OF_BOUNDS);
+        assertCommandFailure(unmatchCommand, model, Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_emptyModel_failure() {
+        UnmatchCommand unmatchCommand = new UnmatchCommand(INDEX_FIRST_PERSON, INDEX_FIRST_JOB);
+        Model emptyModel = new ModelManager(new AddressBook(), new UserPrefs());
+        assertCommandFailure(unmatchCommand, emptyModel, Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_contactNotMatched_failure() {
+        // Confirm that INDEX_FIRST_PERSON is unmatched at the start
+        Person contactToUnmatch = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        assertFalse(contactToUnmatch.isMatchPresent(), "Expected INDEX_FIRST_PERSON"
+                + " to be unmatched at the start of the test.");
```

Comment:
This is not a test on the functionality, this is an assumption. Therefore you should use assertion instead of the `assertFalse()` from `JUnit`.